device_plugins/nvidia_gpu/nvidia_gpu.go

```diff
@@ -0,0 +1,290 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"os/exec"
+	"path"
+	"regexp"
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+
+	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1alpha1"
+)
+
+const (
+	// All NVIDIA GPUs cards should be mounted with nvidiactl and nvidia-uvm
+	// If the driver installed correctly, the 2 devices will be there.
+	nvidiaCtlDevice string = "/dev/nvidiactl"
+	nvidiaUVMDevice string = "/dev/nvidia-uvm"
+	// Optional device.
+	nvidiaUVMToolsDevice string = "/dev/nvidia-uvm-tools"
+	devDirectory                = "/dev"
+	nvidiaDeviceRE              = `^nvidia[0-9]*$`
+
+	// Device plugin settings.
+	devicePluginMountPath = "/device-plugin"
+	kubeletEndpoint       = "kubelet.sock"
+	pluginEndpointPrefix  = "nvidiaGPU"
+	resourceName          = "nvidia.com/GPU"
+	ContainerPathPrefix   = "/usr/local/nvidia"
```

Comment:
note as a next step, we need to also inject ldconfig - relevant Nvidia code [here](https://github.com/NVIDIA/nvidia-docker/blob/3422bca272d64600ca15874d96e4e511b669c7c4/src/nvidia/volumes.go#L234). ideally we should re-use their code.